contrib/Shopify/sarama/sarama.go

```diff
@@ -303,3 +322,57 @@ func getSpanContext(msg *sarama.ProducerMessage) (ddtrace.SpanContext, bool) {
 
 	return spanctx, true
 }
+
+func setProduceCheckpoint(enabled bool, msg *sarama.ProducerMessage, version sarama.KafkaVersion) {
```

Comment:
Looks like there's some overlap here between this function and the equivalent in `confluentinc` does it make sense to pull these out into a `internal/kafkatrace` package to ensure they stay in sync (and reduce duplication)? (see httptrace for an example of this)

Reply:
I'm happy to make the change, but I believe the way each part is checked (message size, version of library, headers) is different for each of sarama & confluent-kafka. So in the end, there isn't really any code I can re-use between the two.